lib/kafka/producer.rb

```diff
@@ -181,6 +180,10 @@ def produce(value, key: nil, topic:, partition: nil, partition_key: nil)
     def send_messages
       attempt = 0
 
+      # Make sure we get metadata for this topic.
+      target_topics = @pending_messages.map(&:topic).uniq
+      @broker_pool.add_target_topics(target_topics)
```

Comment:
An alternative approach would be to successively call e.g. `BrokerPool#add_target_topic` whenever we buffer a message and then e.g. `BrokerPool#ensure_topic_metadata_available!` here. I'm not sure it's an improvement, though, as we'd then need to keep a buffer of topics that we need to fetch metadata from in BrokerPool...